src/codegen/dispatch/solana.rs

```diff
@@ -222,7 +222,21 @@ pub(crate) fn function_dispatch(
     }
 
     match fallback {
-        Some((cfg_no, _)) => {
+        Some((cfg_no, fallback_cfg)) => {
+            let ast_func_no = if let ASTFunction::SolidityFunction(no) = fallback_cfg.function_no {
+                no
+            } else {
+                unreachable!("fallback must be a Solidity function");
+            };
```

Comment:
Can be simplified with a `let-else`

```suggestion
            let ASTFunction::SolidityFunction(ast_func_no) = fallback_cfg.function_no else {
                unreachable!("fallback must be a Solidity function");
            };
```